From 36b86814341d8e5223e13ce7684f4f4d7b37aa8e Mon Sep 17 00:00:00 2001
From: piegames <git@piegames.de>
Date: Mon, 02 Feb 2026 21:25:29 +0100
Subject: [PATCH] deprecated-features/rec-set-dynamic-attrs: Downgrade to warning for now

Looks like too much stuff is breaking on this, so let's make it only a
warning for now to ease transition

Change-Id: I52d50ceb1fe1fbe9f4e28d0aabf3537f7e4d52d1
---

diff --git a/lix/libexpr/parser/state.hh b/lix/libexpr/parser/state.hh
index fc18c5d..e7c3949 100644
--- a/lix/libexpr/parser/state.hh
+++ b/lix/libexpr/parser/state.hh
@@ -231,7 +231,7 @@
 // Added 2025-11-24
 inline void State::recSetDynamicAttrFound(const PosIdx pos)
 {
-    throw ParseError({
+    logWarning({
         .msg = HintFmt(
             "dynamic attributes are not allowed within recursive attrsets, because they would be "
             "evaluated separately from the other recursive attributes. Use %s to disable this "
diff --git a/lix/libutil/deprecated-features/rec-set-dynamic-attrs.md b/lix/libutil/deprecated-features/rec-set-dynamic-attrs.md
index f185532..20d401c 100644
--- a/lix/libutil/deprecated-features/rec-set-dynamic-attrs.md
+++ b/lix/libutil/deprecated-features/rec-set-dynamic-attrs.md
@@ -4,8 +4,8 @@
 timeline:
   - date: 2026-01-30
     release: 2.95.0
-    cls: [4652]
-    message: Introduced as a parser error.
+    cls: [4652, 5067]
+    message: Introduced as a parser warning.
 ---
 Dynamic attrs (attrs with interpolation in the key) are deprecated in `rec` attrsets.
 This is because the recursive dynamics fundamentally do not work with dynamic attributes,
